### PR TITLE
[Fix #11089] Fix an error for `Style/RedundantStringEscape` when using character literals (e.g. `?a`)

### LINKED ISSUE #11089: Style/RedundantStringEscape cop is crashing on some rare inputs
The code as follows cause cop to crash:
```
["one", "two"] * ?.
# => "one.two"
```
--------
## Expected behavior
The cop should not crash. Rubocop probably can suggest to rewrite the code like `["one", "two"].join(".")`, but not crash.

## Actual behavior
`An error occurred while Style/RedundantStringEscape cop was inspecting ...`

## RuboCop version
```
$ [bundle exec] rubocop -V
1.37.0 (using Parser 3.1.2.1, rubocop-ast 1.22.0, running on ruby 3.1.1) [x86_64-linux]
  - rubocop-rails 2.15.2
  - rubocop-rspec 2.12.1
```

### PR DESCRIPTION
**Pull request by ydah**
Fix: #11089

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/redundant_string_escape_spec.rb [hunk 1]
```diff
@@ -279,6 +279,14 @@ def wrap(contents)
     end
   end
 
+  context 'when string literal `?a`' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/redundant_string_escape_spec.rb:282]: I guess that's not specific to `?a`, but applies to any char literal, right? Might be best to update the description accordingly.

**Comment by ydah** [comment]
Thank you so much. I updated this PR.

File: spec/rubocop/cop/style/redundant_string_escape_spec.rb [hunk 2]
```diff
@@ -279,6 +279,14 @@ def wrap(contents)
     end
   end
 
+  context 'when string literal specify one character after `?`' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/redundant_string_escape_spec.rb:282]: I'd shorten this to "when using character literals (e.g. ?a)"

File: spec/rubocop/cop/style/redundant_string_escape_spec.rb [hunk 3]
```diff
@@ -279,6 +279,14 @@ def wrap(contents)
     end
   end
 
+  context 'when using character literals (e.g. `?a`)' do
+    it 'does not register an offense for escaped spaces' do
```
Comment (koic) [inline spec/rubocop/cop/style/redundant_string_escape_spec.rb:283]: "escaped spaces" doesn't seem to be used.

Comment (ydah) [reply spec/rubocop/cop/style/redundant_string_escape_spec.rb]: Oops, indeed it is. Thank you very much.
